tests/php/ORM/ManyManyListTest.php

```diff
@@ -73,14 +72,41 @@ public function testAddCompositedExtraFieldsWithSortColumn0()
 
         // the actual test is that this does not generate an error in the sql.
         $obj->Products()->add($product, [
-            'Reference' => 'Foo'
+            'Reference' => 'Foo',
         ]);
 
         $result = $obj->Products()->First();
         $this->assertEquals('Foo', $result->Reference, 'Basic scalar fields should exist');
         $this->assertEquals('Test Product', $result->Title);
     }
 
+    public function testSetExtraData()
+    {
+        $obj = new ManyManyListTest\ExtraFieldsObject();
+        $obj->write();
+
+        $obj2 = new ManyManyListTest\ExtraFieldsObject();
+        $obj2->write();
+
+        $money = new DBMoney();
+        $money->setAmount(100);
+        $money->setCurrency('USD');
+
+        $obj->Clients()->add($obj2, [
+            'Worth' => $money,
+            'Reference' => 'Foo',
+        ]);
+        $obj->Clients()->setExtraData($obj2->ID, [
+            'Worth' => $money,
+            'Reference' => 'Foo',
+        ]);
```

Comment:
The call to `setExtraData()` sets the same data as `add()`, so I think this test would pass even if `setExtraData()` doesn’t have the desired effect - could you change the data passed to the `setExtraData()` call?

Reply:
Yes, good point. Done.